### PR TITLE
Provisioning API to IBootstrap

### PR DESCRIPTION
**Pull request by rullzer**
Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Review by nickvergessen — APPROVED** [review]

**Review by PVince81 — APPROVED** [review]
:+1:

I'm guessing this is to remove the annoying warnings in the logs ?
Assuming that the other things marked by code scanning won't spam the logs.

**Comment by rullzer** [comment]
> +1
> 
> I'm guessing this is to remove the annoying warnings in the logs ?
> Assuming that the other things marked by code scanning won't spam the logs.

More so Joas can continue with his code :)

**Comment by ChristophWurst** [comment]
please address the inline psalm reports. they seem all legit :)

File: apps/provisioning_api/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -34,20 +34,24 @@
...
-class Application extends App {
+class Application extends App implements IBootstrap {
 	public function __construct(array $urlParams = []) {
 		parent::__construct('provisioning_api', $urlParams);
+	}
 
-		$container = $this->getContainer();
-		$server = $container->getServer();
+	public function register(IRegistrationContext $context): void {
+		$server = $this->getContainer()->getServer();
```
[the remark this answers is not in the document]
Comment (rullzer) [reply apps/provisioning_api/lib/AppInfo/Application.php:52]: All just copied code.
But yeah lets see

**Review by ChristophWurst — APPROVED** [review]

**Comment by nickvergessen** [comment]
Backport worked in https://github.com/nextcloud/server/pull/26044

Not sure what the bot was complaining about

**Pull request merged**